vsintegration/src/unittests/Tests.ProjectSystem.MSBuildUtilities.fs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace UnitTests.Tests.ProjectSystem
+
+open System
+open NUnit.Framework
+open UnitTests.TestLib.Utils
+open Microsoft.VisualStudio.FSharp.ProjectSystem
+
+
+[<TestFixture>]
+type MSBuildUtilitiesTest() = 
+    
+    static let pathSeparatorsPermutations = 
+        let pathSeparators = [""; @"\"; @"\\"] //TODO add '/'
+        [ for x in pathSeparators do 
+            for y in pathSeparators do
+                yield x, y ]
+
+    static let allPathsSeparator l =
+        seq { 
+            for x, y in pathSeparatorsPermutations do
+                for a, b, result in l |> Seq.ofList do
+                    yield a + x, b + y, result
+        }
+
+
+    static member IsSameDirTestCases =
+        [ @"c:\path\to", @"c:\path\to", true
+          @"c:\path\to", @"c:\path\to2", false
+          @"c:\path", @"c:\path", true ]
+        |> allPathsSeparator
+        |> Seq.map (fun (a,b,result) -> TestCaseAttribute(a, b, result))
+
+    [<Test; TestCaseSource("IsSameDirTestCases")>]
+    member public this.IsSameDir a b expected = 
+        Assert.AreEqual(expected, MSBuildHelpers.IsSameDir a b)
+    
+    static member IsSubDirOfTestCases =
+        [ @"c:\path", @"c:\path\to", true
+          @"c:\path\to", @"c:\path\to", false
+          @"c:\path\to", @"c:\path", false
+          @"c:\path\to\another dir", @"c:\path\to\another dir\with relative dir", true ]
+        |> allPathsSeparator
+        |> Seq.map (fun (a,b,result) -> TestCaseAttribute(a, b, result))
+
+    [<Test; TestCaseSource("IsSubDirOfTestCases")>]
+    member public this.IsSubDirOf baseDir dir expected = 
+        Assert.AreEqual(expected, dir |> MSBuildHelpers.IsSubDirOf baseDir)
+
+    static member IsAlreadyRenderedFolderTestCases =
```

Comment:
they are all folders, with inside a file

because for rendere `a\b\c` you need to renderer

    A                                                                            
    +                                                                            
    +-+B                                                                         
    |  +                                                                         
    |  +-+C                                                                      
    |                                                                            
    +-+file.fs                                                                   
    |                                                                            
    +-+B                                                                         
       +                                                                         
       +-+C                                                                      

the first folder rendered two time is a\b